hadoop-tools/hadoop-azure/dev-support/testrun-scripts/testsupport.sh

```diff
@@ -0,0 +1,146 @@
+  if ! [[ $threadcount =~ $numberegex ]] ; then
+    echo "Exiting. The script param (threadcount) should be a number"
+    exit -1
+  fi
+  if [ -z "$scenario" ]; then
+   echo "Exiting. scenario cannot be empty"
+   exit -1
+  fi
+  propertiessize=${#properties[@]}
+  valuessize=${#values[@]}
+  if [ "$propertiessize" -lt 1 ] || [ "$valuessize" -lt 1 ] || [ "$propertiessize" -ne "$valuessize" ]; then
+    echo "Exiting. Both properties and values arrays has to be populated and of same size. Please check for scenario $scenario"
+    exit -1
+  fi
+
+  if ! [ -f "$conffile" ]; then
+    echo "Exiting. $conffile missing"
+    exit -1
+  fi
+}
+
+checkdependencies() {
+  if ! [ "$(command -v pcregrep)" ]; then
+    echo "Exiting. pcregrep is required to run the script."
+    exit -1
+  fi
+  if ! [ "$(command -v xmlstarlet)" ]; then
+    echo "Exiting. xmlstarlet is required to run the script."
+    exit -1
+  fi
+}
+
+changeconf() {
+  xmlstarlet ed -P -L -d "/configuration/property[name='$1']" $conffile
+  xmlstarlet ed -P -L -s /configuration -t elem -n propertyTMP -v "" -s /configuration/propertyTMP -t elem -n name -v "$1" -r /configuration/propertyTMP -v property $conffile
+  if ! xmlstarlet ed -P -L -s "/configuration/property[name='$1']" -t elem -n value -v "$2" $conffile
+  then
+    echo "Exiting. Changing config property failed."
+    exit -1
+  fi
+}
+
+removeproperties() {
+  propertiessize=${#properties[@]}
+  for ((i = 0; i < propertiessize; i++)); do
+    key=${properties[$i]}
+    xmlstarlet ed -P -L -d "/configuration/property[name='$key']" $conffile
+  done
+}
+
+testwithconfs() {
+  propertiessize=${#properties[@]}
+  valuessize=${#values[@]}
+  if [ "$propertiessize" -ne "$valuessize" ]; then
+    echo "Exiting. Number of properties and values differ for $scenario"
+    exit -1
+  fi
+  for ((i = 0; i < propertiessize; i++)); do
+    key=${properties[$i]}
+    val=${values[$i]}
+    changeconf "$key" "$val"
+  done
+  mvn -T 1C -Dparallel-tests=abfs -Dscale -DtestsThreadCount=$threadcount verify >> "$testlogfilename"
+  removeproperties
+}
+
+summary() {
+  {
+    echo ""
+    echo "$scenario"
+    echo "========================"
+    pcregrep -M "$testresultsregex" "$testlogfilename"
+  } >> "$testresultsfilename"
+  printf "\n----- Test results -----\n"
+  pcregrep -M "$testresultsregex" "$testlogfilename"
+
+  secondstaken=$((ENDTIME - STARTTIME))
+  mins=$((secondstaken / 60))
+  secs=$((secondstaken % 60))
+  printf "\nTime taken: %s mins %s secs.\n" "$mins" "$secs"
+  echo "Find test logs for the scenario ($scenario) in: $testlogfilename"
+  echo "Find consolidated test results in: $testresultsfilename"
+  echo "----------"
+}
+
+runtestwithconfs() {
+  validate
+  cp $conffile $bkpconffile
+  if [ -z "$starttime" ]; then
+    checkdependencies
+    mvn clean install -DskipTests
+    starttime=$(date +"%Y-%m-%d_%H-%M-%S")
+    mkdir -p "$logdir"
+    testresultsfilename="$logdir/Test-$starttime-Results.log"
```

Comment:
can you just use .txt as a suffix for text files. IDEs and things often use .log as a format for structured output